var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -403,8 +407,16 @@ def determine_bootstrap_options(self, spec, with_libs, options):
         with open('user-config.jam', 'w') as f:
             # Boost may end up using gcc even though clang+gfortran is set in
             # compilers.yaml. Make sure this does not happen:
-            f.write("using {0} : : {1} ;\n".format(boost_toolset_id,
-                                                   spack_cxx))
+            if not spec.satisfies('%intel') and not spec.satisfies('platform=windows'):
+                # using intel-linux : : spack_cxx in user-config.jam leads to
+                # error: at project-config.jam:12
+                # error: duplicate initialization of intel-linux with the following parameters:  # noqa
+                # error: version = <unspecified>
+                # error: previous initialization at ./user-config.jam:1
```

Comment:
This reintroduces some logic removed in https://github.com/spack/spack/commit/38592b9e4f974254c18ce82e8b7ccca65a07539e - is it possible to preserve the changes in general here and only do this for Windows?